src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,53 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    private static final String BULLDOZER_CLASS_NAME = "core.mate.academy.model.Bulldozer";
+    private static final String EXCAVATOR_CLASS_NAME = "core.mate.academy.model.Excavator";
+    private static final String TRUCK_CLASS_NAME = "core.mate.academy.model.Truck";
+
+    @Override
+    public List getAll(Class<? extends Machine> type) {
+        /*
+        If I parametrize "Class<? extends Machine>", I can't parametrize
+        "List<? extends Machine>". I am getting this information:
+        "getAll(Class<? extends Machine>)' in 'core.mate.academy.service.MachineServiceImpl'
+        clashes with 'getAll(Class<? extends Machine>)'
+        in 'core.mate.academy.service.MachineService'; attempting to use incompatible return type"
+         */
+        String typeName = type.getName();
```

Comment:
You can use `type.getSimpleName()` it will give you a simpler name -> easier constats -> clearer code:)